netplan/cli/commands/set.py

```diff
@@ -159,7 +160,8 @@ def write_file(self, set_tree, name, rootdir='/'):
             # Validate the newly created file, by parsing it via libnetplan
             utils.netplan_parse(tmpp)
             # Valid, move it to final destination
-            os.replace(tmpp, absp)
+            shutil.copy(tmpp, absp)
```

Comment:
wouldn't this loose permissions, ownership, extended attributes, etc? and is that intentional? or should an attempt be made to preserve acls? in which case `shutil.copy2()` should be used.

Reply:
Good catch, thank you! Although, it shouldn't make a big difference, as that file (tmpp) is newly created by that method/process/user. The same method/process/user that would create/copy the file to the new location.
Still, preserving any metadata possible sounds like a good plan!